Sources/Cache/APIs/Sqlite.php

```diff
@@ -65,15 +78,40 @@ public function __construct()
 	public function connect(): bool
 	{
 		$database = $this->cachedir . '/' . 'SQLite3Cache.db3';
-		$this->cacheDB = new SQLite3($database);
-		$this->cacheDB->busyTimeout(1000);
 
-		if (filesize($database) == 0) {
-			$this->cacheDB->exec('CREATE TABLE cache (key text unique, value blob, ttl int);');
-			$this->cacheDB->exec('CREATE INDEX ttls ON cache(ttl);');
+		if (!is_writable($database) || !is_writeable($this->cachedir)) {
+			return false;
 		}
 
-		return true;
+		try {
+			// Did we disable WAL?  That triggers a read only mode, dump the cache.
+			if (file_exists($this->cachedir . '/' . 'SQLite3Cache.db3-wal') && empty(Config::$modSettings['cache_sqlite_wal'])) {
```

Comment:
I think you meant this:

```suggestion
			if (file_exists($this->cachedir . '/' . 'SQLite3Cache.db3-wal') && empty(Config::$cache_sqlite_wal)) {
```

Note also that `Config::$cache_sqlite_wal` won't actually work until you add the necessary content to the Config class, as described in the second part of [this comment](https://github.com/SimpleMachines/SMF/pull/8880#issuecomment-3218502091).